test/byContract/IexecPoco/04_finalize.test.ts

```diff
@@ -733,6 +733,165 @@ describe('IexecPoco2#finalize', async () => {
+            .connect(worker1)
+            .contribute(
+                taskId,
+                resultHash,
+                resultSeal,
+                emptyEnclaveAddress,
+                emptyEnclaveSignature,
+                schedulerSignature,
+            )
+            .then((tx) => tx.wait());
+        await iexecPoco
+            .connect(worker1)
+            .reveal(taskId, callbackResultDigest)
+            .then((tx) => tx.wait());
+        const task = await iexecPoco.viewTask(taskId);
+        await time.setNextBlockTimestamp(task.revealDeadline);
+        const { resultsCallback } = buildResultCallbackAndDigest(567);
+        await expect(
+            iexecPocoAsScheduler.finalize(taskId, results, resultsCallback),
+        ).to.be.revertedWithoutReason();
+    });
+
+    it('Should finalize task when result callback is invalid', async () => {
+        const oracleConsumerInstance = await new TestClient__factory()
+            .connect(anyone)
+            .deploy()
+            .then((contract) => contract.deployed());
+        const orders = buildOrders({
+            assets: ordersAssets,
+            requester: requester.address,
+            prices: ordersPrices,
+            callback: appAddress,
+        });
+
+        const { dealId, taskId, taskIndex } = await iexecWrapper.signAndMatchOrders(
+            ...orders.toArray(),
+        );
+        await iexecPoco.initialize(dealId, taskIndex).then((tx) => tx.wait());
+        const workerTaskStake = await iexecPoco
+            .viewDeal(dealId)
+            .then((deal) => deal.workerStake.toNumber());
+        const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+            taskId,
+            callbackResultDigest,
+            worker1,
+        );
+        const schedulerSignature = await buildAndSignContributionAuthorizationMessage(
+            worker1.address,
+            taskId,
+            emptyEnclaveAddress,
+            scheduler,
+        );
+        await iexecWrapper.depositInIexecAccount(worker1, workerTaskStake);
+        await iexecPoco
+            .connect(worker1)
+            .contribute(
+                taskId,
+                resultHash,
+                resultSeal,
+                emptyEnclaveAddress,
+                emptyEnclaveSignature,
+                schedulerSignature,
+            )
+            .then((tx) => tx.wait());
+        await iexecPoco
+            .connect(worker1)
+            .reveal(taskId, callbackResultDigest)
+            .then((tx) => tx.wait());
+        const task = await iexecPoco.viewTask(taskId);
+        await time.setNextBlockTimestamp(task.revealDeadline);
+        await expect(iexecPocoAsScheduler.finalize(taskId, results, resultsCallback))
+            .to.emit(iexecPoco, 'TaskFinalize')
+            .withArgs(taskId, hexResults)
+            .to.not.emit(oracleConsumerInstance, 'GotResult');
+    });
+
+    it('Should finalize task when callback is EOA', async () => {
```

Comment:
I would move this closer to "should tests" maybe below "Should finalize task of deal payed by requester (no callback, no dataset)" (?)

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/145/commits/36a69a090fb1dba3d2d70e09565b30cb6a3841b1)

Reply:
Updated [here](https://github.com/iExecBlockchainComputing/PoCo/pull/145/files/5cbafb6352e5d795ab31a06ac809ab89bdc29a95..e991668003ec1dafb1dd6ced4e95b3fd50650053)